dataInterface/movies.js

```diff
@@ -118,13 +141,35 @@ module.exports.updateById = async (movieId, newObj) => {
   const result = await movies.updateOne(filter, updateRules);
 
   if(result.modifiedCount != 1){
-    return {error: `Something went wrong. ${result.modifiedCount} movies were updated. Please try again.`}
+    return {error: `Something went wrong. ${result.modifiedCount} movies were not updated. Please try again.`}
   };
 
   const updatedMovie = module.exports.getById(movieId);
   return updatedMovie;
 }
 
+module.exports.updateCommentById = async (movieID, commentID, text) =>{
+  const database = client.db(databaseName);
+  const comments = database.collection(commCollName);
+  const query = { movie_id: ObjectId(movieID)};
+
+  let commentCursor = await comments.find(query);
+  let commentArray = commentCursor.toArray()
+  if(commentArray.length >=1){
+    for(let i =0 ; i<=commentArray.length; i++){
+      if(i._id === commentID){
+        return i
+      }else{
+        return {error: "comment does not exist"}
+      }
+    }
+    console.log(comment)
+  }else{
+    return {error: "Comments cannot be found"}
+  }
```

Comment:
Finding the comment objects with the given movieId isn't necessary. The goal is to update just a single record, which can be found by querying for its _id field, then applying the updates. I'd expect this to look like: 
```
  const updateRules = {
    $set: {"text" : text }
  };
  const filter = { _id: ObjectId(commentID) };
  const result = await comments.updateOne(filter, updateRules);
```